### PR TITLE
Enable TokenBrokerResolver for mTLS

### PR DESCRIPTION
**Pull request by liga-oz**

### CODE REVIEW

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public TokenBrokerResolver(XsuaaServiceConfiguration configuration, Cache tokenC
 			AuthenticationInformationExtractor authenticationConfig) {
 		this.configuration = configuration;
 		this.tokenCache = tokenCache;
-		this.tokenBroker = new UaaTokenBroker(tokenService);
+		this.oAuth2TokenService = tokenService;
```
Comment (nenaraab) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:101]: @liga-oz can you please check that UaaTokenBroker class is marked as deprecated?

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java [hunk 2]
```diff
@@ -296,22 +312,4 @@ public AuthenticationInformationExtractor getAuthenticationConfig() {
...
-
-	/**
-	 * Returns the subdomain claim "zdn" of a given OIDC access token (JWT bearer).
-	 *
-	 * @param encodedJwtToken
-	 *            - the token to check the scope for.
-	 * @return subdomain or null.
-	 */
-	@Nullable
-	private String parseSubdomainFromOIDCToken(String encodedJwtToken) {
```
Comment (nenaraab) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:308]: thanks for deleting dead code `parseSubdomainFromOIDCToken`

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java [hunk 3]
```diff
@@ -212,10 +212,18 @@ private String getBrokerToken(AuthenticationMethod credentialType, String authHe
 					logger.debug("return (basic) access token for {} from cache", cacheKey);
 					return cachedToken;
 				} else {
-					String token = tokenBroker.getAccessTokenFromPasswordCredentials(oauthTokenUrl,
-							clientIdentity.getId(),
-							clientIdentity.getSecret(), userCredentialsFromHeader.getId(),
-							userCredentialsFromHeader.getSecret());
+					String token;
+					if (tokenBroker == null) {
```
Comment (nenaraab) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:216]: couldn't you check if oAuth2TokenService != null

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java [hunk 4]
```diff
@@ -233,10 +241,18 @@ private String getBrokerToken(AuthenticationMethod credentialType, String authHe
...
+						token = oAuth2TokenService.retrieveAccessTokenViaClientCredentialsGrant(
+								URI.create(oauthTokenUrl), clientIdentity, null, null,
+								null, false).getAccessToken();
+					} else {
+						// only when deprecated constructors are used, for backward compatibility
+						token = tokenBroker
+								.getAccessTokenFromClientCredentials(oauthTokenUrl, clientCredentialsFromHeader.getId(),
+										clientCredentialsFromHeader.getSecret());
+						tokenCache.put(cacheKey, token);
```
Comment (nenaraab) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:254]: tokenCache must be filled outside the if/else clause when token is not null

**Review by nenaraab — CHANGES_REQUESTED** [review]

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java (hunk 1 above)
Comment (liga-oz) [reply spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:101]: @nenaraab yes, it is marked as deprecated

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java (hunk 4 above)
Comment (liga-oz) [reply spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/extractor/TokenBrokerResolver.java:254]: good point 👍🏻

**Review by nenaraab — APPROVED** [review]

**Comment by nenaraab** [comment]
:sparkling_heart:

**Comment by nenaraab** [comment]
closes #548

**Pull request merged**